File: libs/eave-stdlib-py/src/eave/stdlib/core_api/models/github_repos.py
import enum
from typing import Optional
import uuid

from eave.stdlib.core_api.models import BaseResponseModel

from eave.stdlib.core_api.models import BaseInputModel


class GithubRepoFeature(enum.StrEnum):
    API_DOCUMENTATION = "api_documentation"
    INLINE_CODE_DOCUMENTATION = "inline_code_documentation"
    ARCHITECTURE_DOCUMENTATION = "architecture_documentation"


class GithubRepoFeatureState(enum.StrEnum):
    DISABLED = "disabled"
    ENABLED = "enabled"
    PAUSED = "paused"


class GithubRepo(BaseResponseModel):
    team_id: uuid.UUID
    id: uuid.UUID
    github_installation_id: uuid.UUID
    external_repo_id: str
    display_name: Optional[str]
    api_documentation_state: GithubRepoFeatureState
    inline_code_documentation_state: GithubRepoFeatureState
    architecture_documentation_state: GithubRepoFeatureState


class GithubRepoCreateInput(BaseInputModel):
    external_repo_id: str
    display_name: str
    api_documentation_state: Optional[GithubRepoFeatureState] = None
    inline_code_documentation_state: Optional[GithubRepoFeatureState] = None
    architecture_documentation_state: Optional[GithubRepoFeatureState] = None


class GithubRepoRefInput(BaseInputModel):
    id: uuid.UUID


class GithubRepoListInput(BaseInputModel):
    external_repo_id: str


class GithubRepoUpdateValues(BaseInputModel):
    api_documentation_state: Optional[GithubRepoFeatureState] = None
    inline_code_documentation_state: Optional[GithubRepoFeatureState] = None
    architecture_documentation_state: Optional[GithubRepoFeatureState] = None


class GithubRepoUpdateInput(BaseInputModel):
    id: uuid.UUID
    new_values: GithubRepoUpdateValues


class GithubReposDeleteInput(BaseInputModel):
    id: uuid.UUID


class GithubReposFeatureStateInput(BaseInputModel):
    feature: GithubRepoFeature
    state: GithubRepoFeatureState

File: libs/eave-stdlib-ts/src/core-api/models/github-repos.ts
import { UUID } from "../../types.js";

export enum GithubRepoFeature {
  API_DOCUMENTATION = "api_documentation",
  INLINE_CODE_DOCUMENTATION = "inline_code_documentation",
  ARCHITECTURE_DOCUMENTATION = "architecture_documentation",
}

export enum GithubRepoFeatureState {
  DISABLED = "disabled",
  ENABLED = "enabled",
  PAUSED = "paused",
}

export type GithubRepo = {
  team_id: UUID;
  id: UUID;
  github_installation_id: UUID;
  external_repo_id: string;
  display_name: string | null;
  api_documentation_state: GithubRepoFeatureState;
  inline_code_documentation_state: GithubRepoFeatureState;
  architecture_documentation_state: GithubRepoFeatureState;
};

export type GithubRepoCreateInput = {
  external_repo_id: string;
  display_name: string;
  api_documentation_state?: GithubRepoFeatureState;
  inline_code_documentation_state?: GithubRepoFeatureState;
  architecture_documentation_state?: GithubRepoFeatureState;
};

export type GithubRepoRefInput = {
  id: UUID;
};

export type GithubRepoListInput = {
  external_repo_id: string;
};

export type GithubRepoUpdateValues = {
  api_documentation_state?: GithubRepoFeatureState;
  inline_code_documentation_state?: GithubRepoFeatureState;
  architecture_documentation_state?: GithubRepoFeatureState;
};

export type GithubRepoUpdateInput = {
  id: UUID;
  new_values: GithubRepoUpdateValues;
};

export type GithubReposDeleteInput = {
  id: UUID;
};

export type GithubReposFeatureStateInput = {
  feature: GithubRepoFeature;
  state: GithubRepoFeatureState;
};
